pilot/pkg/model/telemetry.go

```diff
@@ -1052,3 +1053,47 @@ func generateStatsConfig(class networking.ListenerClass, metricsCfg telemetryFil
 func disableHostHeaderFallback(class networking.ListenerClass) bool {
 	return class == networking.ListenerClassSidecarInbound || class == networking.ListenerClassGateway
 }
+
+// resourceAttributes return semantic attributes follow OpenTelemetry specification
+// see more details here: https://opentelemetry.io/docs/reference/specification/resource/semantic_conventions/
+func resourceAttributes(proxy *Proxy) *otlpcommon.KeyValueList {
+	// TODO(zirain): generate resource attributes for VM
+	if proxy.IsVM() {
+		return nil
+	}
+
+	podName, namespace := extractProxyID(proxy)
+
+	return &otlpcommon.KeyValueList{
+		Values: []*otlpcommon.KeyValue{
+			{
+				Key:   "k8s.cluster.name",
+				Value: &otlpcommon.AnyValue{Value: &otlpcommon.AnyValue_StringValue{StringValue: proxy.Metadata.ClusterID.String()}},
+			},
+			{
+				Key: "k8s.service.name",
+				// this seems a little hack, but in istio it works, we do the samething for `serviceCluster` in tracing
+				Value: &otlpcommon.AnyValue{Value: &otlpcommon.AnyValue_StringValue{StringValue: proxy.XdsNode.Cluster}},
+			},
+			{
+				Key:   "k8s.pod.namespace",
```

Comment:
this doesn't seem to be an OTEL one either?

Reply:
`k8s.namespace.name`. We should be aligning with https://github.com/istio/istio/blob/373fc89518c986c9f48ed3cd891930da6fdc8628/pilot/pkg/networking/core/v1alpha3/listener.go#L1653